include/deal.II/lac/affine_constraints.templates.h

```diff
@@ -2806,6 +2806,16 @@ AffineConstraints<number>::distribute(VectorType &vec) const
 {
   Assert(sorted == true, ExcMatrixNotClosed());
 
+#ifdef DEAL_II_TRILINOS_WITH_TPETRA
+  if constexpr (std::is_same<
+                  VectorType,
+                  LinearAlgebra::TpetraWrappers::Vector<number>>::value)
```

Comment:
I think you can make this generic over all vector types (and then also get rid of the `#ifdef` using `internal::is_supported_operation`. I think this would look like this:
```suggestion
  template <typename V>
  using is_compressed_op = decltype(std::declval<V>().is_compressed());
  
  if constexpr (internal::is_supported_operation<is_compressed_op, VectorType>)
```

Reply:
I will give it a try.

Reply:
What's the error message you get? Either way, if you can't get it to work, we can leave it as is and I can give it a try myself later.

Reply:
I apologize for not providing any details earlier.

I added this patch again so you can look at it yourself.
The Error message is:
````
/__w/dealii/dealii/include/deal.II/lac/affine_constraints.templates.h:2809:3: error: a template declaration cannot appear at block scope
 2809 |   template <typename V>
      |   ^~~~~~~~
```

Reply:
Yes, you have to provide the template alias outside of function scope, preferrably at namespace scope.

Reply:
Thank you! That fixed it.

Reply:
I just squashed all commits into one. 

If I should move the line
```
  template <typename V>
  using is_compressed_op = decltype(std::declval<V>().is_compressed());
```
please let me know.